lib/Valiemon/Attributes/AnyOf.pm

```diff
@@ -0,0 +1,46 @@
+package Valiemon::Attributes::AnyOf;
+use strict;
+use warnings;
+use utf8;
+use parent qw(Valiemon::Attributes);
+
+use Carp qw(croak);
+use List::MoreUtils qw(any);
+
+use Valiemon::Context;
+
+sub attr_name { 'anyOf' }
+
+sub is_valid {
+    my ($class, $context, $schema, $data) = @_;
+
+    my $anyOf = $schema->{anyOf};
+    if (ref $anyOf ne 'ARRAY' || scalar @$anyOf < 1) {
+        croak sprintf '`anyOf` must be an array and have at least one element at %s', $context->position
+    }
+
+    my $idx = 0;
+    my $errors = [];
+    # Clone context to avoid influencing sub_validator's error to main context.
+    my $is_valid = any {
+        my $clone_context = Valiemon::Context->clone_from($context);
+        my ($is_partially_valid, $error) = $clone_context->in_attr($class, sub {
+            my $sub_v = $clone_context->sub_validator($_);
+            $clone_context->in(
+                $idx++, sub { $sub_v->validate($data, $clone_context); }
+            );
+        });
+        # If an error occured keep it so that main context will return it as main context's validation error.
+        if ($error) {
+            push @$errors, $error;
+        }
+        $is_partially_valid;
+    } @$anyOf;
+
+    unless ($is_valid) {
+        return (0, $errors->[0]);
+    }
+    return 1;
```

Comment:
I think it can be more simple. How about this?

```perl
sub is_valid {
    my ($class, $context, $schema, $data) = @_;

    my $anyOf = $schema->{anyOf};
    if (ref $anyOf ne 'ARRAY' || scalar @$anyOf < 1) {
        croak sprintf '`anyOf` must be an array and have at least one element at %s', $context->position
    }

    $context->in_attr($class, sub {
        any {
            my $clone_context = Valiemon::Context->clone_from($context);
            $clone_context->sub_validator($_)->validate($data, $clone_context);
        } @$anyOf;
    });
}
```

This impl returns same response like this.
https://github.com/pokutuna/p5-Valiemon/pull/14#discussion_r60171041 

```perl
    my ($res, $err);
    my $v = Valiemon->new({
        anyOf => [
            { properties => { name => { type => "string" } }, required => ['name'] },
            { properties => { age => { type => "integer" } } },
        ],
    });
# ...
    ($res, $err) = $v->validate({ age => 'secret' });
    ok !$res, 'failed both schema validation';
    is $err->position, '/anyOf';
    use Data::Dumper; warn Dumper $err;
```

```perl
 $VAR1 = bless( {
                 'attribute' => 'Valiemon::Attributes::AnyOf',
                 'actual' => {
                               'age' => 'secret'
                             },
                 'expected' => {
                                 'anyOf' => [
                                              {
                                                'required' => [
                                                                'name'
                                                              ],
                                                'properties' => {
                                                                  'name' => {
                                                                              'type' => 'string'
                                                                            }
                                                                }
                                              },
                                              {
                                                'properties' => {
                                                                  'age' => {
                                                                             'type' => 'integer'
                                                                           }
                                                                }
                                              }
                                            ]
                               },
                 'position' => '/anyOf'
               }, 'Valiemon::ValidationError' );
```